packages/@aws-cdk/aws-lambda/lib/dlq.ts

```diff
@@ -0,0 +1,22 @@
+import { IResource } from '@aws-cdk/core';
+import { IFunction } from './function-base';
+
+/**
+ * A destination configuration
+ */
+export interface DlqDestinationConfig {
+  /**
+   * The Amazon Resource Name (ARN) of the destination resource
+   */
+  readonly destination: string;
+}
+
+/**
+ * A DLQ for an event source
+ */
+export interface IEventSourceDlq {
+  /**
+   * Returns the DLQ destination config of the DLQ
+   */
+  bind(target: IResource, targetHandler: IFunction): DlqDestinationConfig;
```

Comment:
Why `IResource` instead of `IEventSourceMapping` ?

Reply:
Actually, I've gone ahead and added an `IEventSourceMapping` interface to the APIs. You should be able to switch to these for your `bind()` API in your `Dlq` classes.

Let me know if you have any issues.